include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2972,46 +2794,37 @@ namespace internal
 
 
 
-  template <int dim, typename VectorizedArrayType>
+  template <int dim, typename Number>
   struct FEFaceEvaluationImplEvaluateSelector
   {
     template <int fe_degree, int n_q_points_1d>
     static bool
-    run(const unsigned int                                         n_components,
-        const MatrixFreeFunctions::ShapeInfo<VectorizedArrayType> &data,
-        const VectorizedArrayType *                                values_array,
-        VectorizedArrayType *                                      values_quad,
-        VectorizedArrayType *         gradients_quad,
-        VectorizedArrayType *         hessians_quad,
-        VectorizedArrayType *         scratch_data,
-        const bool                    evaluate_values,
-        const bool                    evaluate_gradients,
-        const bool                    evaluate_hessians,
-        const unsigned int            face_no,
-        const unsigned int            subface_index,
-        const unsigned int            face_orientation,
-        const Table<2, unsigned int> &orientation_map)
+    run(const unsigned int                     n_components,
+        const EvaluationFlags::EvaluationFlags evaluation_flag,
+        const Number *                         values_dofs,
+        FEEvaluationBaseData<dim, typename Number::value_type, true, Number>
+          &eval)
     {
+      const auto &       data             = eval.get_shape_info();
+      const auto &       shape_data       = data.data.front();
+      const unsigned int face_no          = eval.get_face_no();
+      const unsigned int face_orientation = eval.get_face_orientation();
+
       if (data.element_type == MatrixFreeFunctions::tensor_none)
         {
           const unsigned int n_dofs     = data.dofs_per_component_on_cell;
           const unsigned int n_q_points = data.n_q_points_faces[face_no];
-          const auto         shape_info = data.data.front();
 
-          using Eval = EvaluatorTensorProduct<evaluate_general,
-                                              1,
-                                              0,
-                                              0,
-                                              VectorizedArrayType,
-                                              VectorizedArrayType>;
+          using Eval =
+            EvaluatorTensorProduct<evaluate_general, 1, 0, 0, Number, Number>;
 
-          if (evaluate_values)
+          if (evaluation_flag & EvaluationFlags::values)
             {
               const auto shape_values =
-                &shape_info.shape_values_face(face_no, face_orientation, 0);
+                &shape_data.shape_values_face(face_no, face_orientation, 0);
```

Comment:
`shape_info`/`shape_data`/`data`: we have noticeable name confusion throughout the code stack...

Reply:
Let me see if I can unify those.